### PR TITLE
[ADD] 8.0 - report_xlsx

### PR DESCRIPTION
**Pull request by sebalix**
This is a backport of https://github.com/OCA/reporting-engine/pull/38 from @adrienpeiffer (migrated with the instructions found here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#if-the-module-doesnt-exist-in-the-90-branch).
It fixes https://github.com/OCA/reporting-engine/issues/49 and is needed to solve https://github.com/OCA/account-financial-reporting/issues/8

### CODE REVIEW

**Comment by sebalix** [comment]
Runbot is green, and ready for "review" (it is the same code that the v9 version).

File: report_xlsx/README.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        string="Print to XLSX"
+        report_type="xlsx"
+        name="res.partner.xlsx"
+        file="res.partner.xlsx"
+        attachment_use="False"
+    />
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/143/9.0
```
Comment (pedrobaeza) [inline report_xlsx/README.rst]: Change this to 8.0

File: report_xlsx/README.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/reporting-engine/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/reporting-engine/issues/new?body=module:%20report_xlsx%0Aversion:%209.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline report_xlsx/README.rst]: 8.0

**Comment by pedrobaeza** [comment]
You can put the icon used in XLSX files for the module, as the report_xls do. And please squash your commits in one, as they are minimal.

File: report_xlsx/README.rst (hunk 1 above)
Comment (sebalix) [inline report_xlsx/README.rst]: Question: where do we get this ID 143? I was looking for that in https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md but didn't find this information.

Comment (pedrobaeza) [inline report_xlsx/README.rst]: It's in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt. In the original README template you can see the link: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

Comment (sebalix) [inline report_xlsx/README.rst]: Thank you, I missed that.

**Comment by sebalix** [comment]
@pedrobaeza you mean the same icon used by report_xls? https://github.com/OCA/reporting-engine/blob/9.0/report_xls/static/description/icon.png

**Comment by sebalix** [comment]
Done.

**Comment by pedrobaeza** [comment]
> @pedrobaeza you mean the same icon used by report_xls? https://github.com/OCA/reporting-engine/blob/9.0/report_xls/static/description/icon.png

I suppose there's a new one corresponding to last Excel versions that fits with XLSX format, isn't it? Or maybe an special icon not for Excel, but for the mimetype XLSX (like this search: https://www.google.es/search?q=xlsx+icon&safe=off&client=ubuntu&hs=bDY&channel=fs&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjFv-W6qq3MAhViF8AKHQKoBbIQ_AUIBygB&biw=1700&bih=901)

**Comment by sebalix** [comment]
Yep I already search that kind of icon for the XLSX format some days ago but did'nt find one with a correct license. I will take a look again.

**Comment by sebalix** [comment]
Well, just passed half an hour to find out an icon compatible with the AGPL license, without any success. For now I just copied the one from `report_xls`, but even this one I'm not sure if it is really free.

**Comment by pedrobaeza** [comment]
The icon is not ruled under AGPL. The possible open license to apply is Creative Commons, but in this case, I think we can use it without problems under the fair usage policy: http://libguides.mit.edu/usingimages

**Comment by sebalix** [comment]
There are several Creative Commons, but all CC icons found have a NC clause.
So I pick one and update the PR? This subject has already been stated before in OCA?

**Comment by sebalix** [comment]
Okay, this one is a CC with commercial use allowed: http://www.icons101.com/icon/id_67712/setid_2096/Boxed_Metal_by_Martin/xlsx
PR updated with this new icon.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by pedrobaeza** [comment]
Thanks!

File: report_xlsx/README.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Usage
+=====
+
+An example of XLSX report for partners:
+
+A python class ::
+
+    from openerp.addons.report_xlsx.report.report_xlsx import ReportXlsx
+
+    class partner_xlsx(ReportXlsx):
```
Comment (bealdav) [inline report_xlsx/README.rst:27]: a complete detail here but why not `class PartnerXlsx()` here ? However, It must not prevent merge

File: report_xlsx/report/report_xlsx.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+from openerp.report.report_sxw import report_sxw
+from openerp.api import Environment
+
+import logging
+_logger = logging.getLogger(__name__)
+
+try:
+    import xlsxwriter
+except ImportError:
+    _logger.debug('Can not import xlsxwriter`.')
```
Comment (bealdav) [inline report_xlsx/report/report_xlsx.py:16]: could be `_logger.warning()`, isn't it ?

**Comment by bealdav** [comment]
:+1: thanks a lot

File: report_xlsx/report/report_xlsx.py (hunk 4 above)
Comment (pedrobaeza) [inline report_xlsx/report/report_xlsx.py:16]: No, because that implies a warning in runbot and thus a red cross in GitHub CIs

Comment (bealdav) [inline report_xlsx/report/report_xlsx.py:16]: ok you're right

**Comment by mourad-ehm** [comment]
:+1:  Code review

**Comment by hparfr** [comment]
:+1:

**Pull request merged**